File: Homework/JavaScript-Fundamentals/05. Conditional Statements/quadratic-equation.js
function solve(args) {
    var a = +args[0];
    var b = +args[1];
    var c = +args[2];

    var firstRoot;
    var secondRoot;

    var sqrtPart = (b * b) - (4 * a * c);

    if (sqrtPart < 0) {
        console.log('no real roots');
    } else {
        firstRoot = (-b + Math.sqrt(sqrtPart)) / (2 * a);
        secondRoot = (-b - Math.sqrt(sqrtPart)) / (2 * a);

        if (sqrtPart == 0) {
            console.log('x1=x2=' + firstRoot.toFixed(2));
        } else {
            console.log('x1=' +
                Math.min(firstRoot, secondRoot).toFixed(2) +
                '; ' +
                'x2=' +
                Math.max(firstRoot, secondRoot).toFixed(2));
        }
    }
}

File: Homework/CSharp-Part-1/04. Console-In-and-Out/QuadraticEquation.cs
using System;

class QuadraticEquation
{
    static void Main()
    {
        double a = double.Parse(Console.ReadLine());
        double b = double.Parse(Console.ReadLine());
        double c = double.Parse(Console.ReadLine());

        double sqrtPart = (b * b) - (4 * a * c);
        bool hasNoRoots = sqrtPart < 0;
        // if no roots set 0 to avoid Exception
        sqrtPart = hasNoRoots ? 0 : sqrtPart;

        double firstRoot = (-b + Math.Sqrt(sqrtPart)) / (2 * a);
        double secondRoot = (-b - Math.Sqrt(sqrtPart)) / (2 * a);

        // swap to meet problem requirement: <Print the smaller root on the first line>
        bool isFirstGreater = firstRoot > secondRoot;
        double tempRoot = firstRoot;
        firstRoot = isFirstGreater ? secondRoot : firstRoot;
        secondRoot = isFirstGreater ? tempRoot : secondRoot;

        bool hasOneRoot = (sqrtPart == 0);
        string oneRootFormat = "{0:0.00}";
        string twoRootFormat = "{0:0.00}\n{1:0.00}";
        string result = hasOneRoot ? 
            String.Format(oneRootFormat, firstRoot) :
            String.Format(twoRootFormat, firstRoot, secondRoot);

        Console.WriteLine(hasNoRoots ? "no real roots" : result);
    }
}